File: Reverse Linked List II/Reverse Linked List II.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode reverseBetween(ListNode head, int m, int n) {
        if(head==null || head.next==null) return head;

        ListNode newhead = new ListNode(0);
        newhead.next = head;
        ListNode p = newhead;
        int k = 1;
        while(k++<m) p = p.next;
        ListNode p1 = p.next;
        ListNode temp = p1;
        while(m++<n) {
            temp = p1.next;
            p1.next = temp.next;
            temp.next = p.next;
            p.next = temp;
        }
        return newhead.next;
    }
}

File: Reverse Linked List II/Reverse Linked List II .py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param head, a ListNode
    # @param m, an integer
    # @param n, an integer
    # @return a ListNode
    def reverseBetween(self, head, m, n):
        if not head or not head.next:
            return head
        else:
            newhead=ListNode(0)
            newhead.next=head
            p=newhead
            k=1
            while k<m:
                k=k+1
                p=p.next
            p1=p.next
            temp=p1
            while m<n:
                m=m+1
                temp=p1.next
                p1.next=temp.next
                temp.next=p.next
                p.next=temp
            return newhead.next
